pypowerwall/__main__.py

```diff
@@ -11,76 +11,71 @@
 
 """
 
-# Modules
-import pypowerwall
-import sys
+import argparse
 import os
-from . import scan
-from . import cloud
+import sys
+
+# Modules
+from pypowerwall import version
 
 # Global Variables
 AUTHFILE = ".pypowerwall.auth"
 authpath = os.getenv("PW_AUTH_PATH", "")
+
 timeout = 1.0
 hosts = 30
 state = 0
 color = True
 ip = None
 email = None
 
-for i in sys.argv:
-    if(i==sys.argv[0]):
-        continue
-    elif(i.lower() == "scan"):
-        state = 0
-    elif(i.lower() == "setup"):
-        state = 1
-    elif(i.lower() == "-nocolor"):
-        color = False
-    elif(i.lower()[0:4] == "-ip="):
-        ip = i[4:]
-    elif(i.lower()[0:7] == "-hosts="):
-        try:
-            hosts = int(i[7:])
-        except:
-            state = 2
-    elif(i.lower()[0:7] == "-email="):
-        email = i[7:]
-    else:
-        try:
-            timeout = float(i)
-        except:
-            state = 2
+# Setup parser and groups
+p = argparse.ArgumentParser(prog="PyPowerwall", description=f"PyPowerwall Module v{version}")
+subparsers = p.add_subparsers(dest="command", title='commands (run <command> -h to see usage information)',
+                              required=True)
+setup_args = subparsers.add_parser("setup", help='Setup Tesla Login for Cloud Mode access')
+setup_args.add_argument("-email", type=str, default=email, help="Email address for Tesla Login.")
 
-# State 0 = Run Scan
-if(state == 0):
-    scan.scan(color, timeout, hosts, ip)
+scan_args = subparsers.add_parser("scan", help='Scan local network for Powerwall gateway')
+scan_args.add_argument("-timeout", type=float, default=timeout,
+                       help=f"Seconds to wait per host [Default={timeout:.1f}]")
+scan_args.add_argument("-nocolor", action="store_true", default=not color,
+                       help="Disable color text output.")
+scan_args.add_argument("-ip", type=str, default=ip, help="(Scan option) IP address within network to scan.")
```

Comment:
The "(Scan option) " text prefix in the help line for "-ip" argument should be removed now.

Reply:
Good catch! Ad always.